platforms/software/reporting/src/main/java/org/gradle/api/reporting/internal/ReportUtilities.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.reporting.internal;
+
+import org.gradle.api.Project;
+
+public interface ReportUtilities {
+    static String getApiDocTitleFor(Project project) {
```

Comment:
🤔 This method seems to be called mostly to create a provider.  Why not just have it return a `Provider<String>` directly?

Reply:
This is only used with old school convention mapping right now, so it would need to return a `Callable<String>`. 

I'd rather not upset this too much because what we're really missing is a `Provider<String>` of `project.version`, which would let us write this without explicitly creating a Provider.